src/main/java/edu/kis/powp/jobs2d/drivers/visitors/DeepCopyVisitor.java

```
@@ -0,0 +1,233 @@
+    }
+
+    @Override
+    public void visit(ComplexDriver driver) {
+        ComplexDriver copy = new ComplexDriver();
+
+        for (VisitableJob2dDriver child : driver.getChildren()) {
+            child.accept(this);
+            VisitableJob2dDriver childCopy = this.getCopiedDriver();
+            if (childCopy != null) {
+                copy.add(childCopy);
+            }
+        }
+
+        this.copiedDriver = copy;
+    }
+
+    @Override
+    public void visit(InformativeLoggerDriver driver) {
+        this.copiedDriver = new InformativeLoggerDriver();
+    }
+
+    @Override
+    public void visit(LineDriverAdapter driver) {
+        try {
+            DrawPanelController drawController = getDrawController(driver);
+            ILine line = getLine(driver);
+            String name = getName(driver);
+
+            LineDriverAdapter copy = new LineDriverAdapter(
+                    drawController,
+                    LineFactory.getBasicLine(),
+                    name != null ? name + " (Copy)" : "Copied Adapter"
+            );
+
+            try {
+                int startX = getStartX(driver);
+                int startY = getStartY(driver);
+                copy.setPosition(startX, startY);
+            } catch (Exception e) {
+                copy.setPosition(0, 0);
+            }
+
+            this.copiedDriver = copy;
+
+        } catch (Exception e) {
+            LineDriverAdapter copy = new LineDriverAdapter(
+                    null,
+                    LineFactory.getBasicLine(),
+                    "Copied Adapter"
+            );
+            copy.setPosition(0, 0);
+            this.copiedDriver = copy;
+        }
+    }
+
+    @Override
+    public void visit(AbstractDecorator decorator) {
+        VisitableJob2dDriver innerDriver = decorator.getDriver();
+        if (innerDriver != null) {
+            innerDriver.accept(this);
+        } else {
+            this.copiedDriver = new InformativeLoggerDriver();
+        }
```

This doesn't make sense. It would be better to throw an exception